execute/window_test.go

```diff
@@ -208,6 +208,54 @@ func TestWindow_GetEarliestBounds(t *testing.T) {
 				Stop:  execute.Time(5*time.Minute + 30*time.Second),
 			},
 		},
+		{
+			name: "truncate before offset",
+			w: MustWindow(
+				values.ConvertDurationNsecs(5*time.Second),
+				values.ConvertDurationNsecs(5*time.Second),
+				values.ConvertDurationNsecs(2*time.Second), false),
+			t: execute.Time(1 * time.Second),
+			want: execute.Bounds{
+				Start: execute.Time(-3 * time.Second),
+				Stop:  execute.Time(2 * time.Second),
+			},
+		},
+		{
+			name: "truncate after offset",
+			w: MustWindow(
+				values.ConvertDurationNsecs(5*time.Second),
+				values.ConvertDurationNsecs(5*time.Second),
+				values.ConvertDurationNsecs(2*time.Second), false),
+			t: execute.Time(3 * time.Second),
+			want: execute.Bounds{
+				Start: execute.Time(2 * time.Second),
+				Stop:  execute.Time(7 * time.Second),
+			},
+		},
+		{
+			name: "truncate before calendar offset",
+			w: MustWindow(
+				values.ConvertDurationMonths(5),
+				values.ConvertDurationMonths(5),
+				values.ConvertDurationMonths(2), true),
+			t: mustParseTime("1970-02-01T00:00:00Z"),
+			want: execute.Bounds{
+				Start: mustParseTime("1969-10-01T00:00:00Z"),
+				Stop:  mustParseTime("1970-03-01T00:00:00Z"),
+			},
+		},
+		{
+			name: "truncate after calendar offset",
+			w: MustWindow(
+				values.ConvertDurationMonths(5),
+				values.ConvertDurationMonths(5),
+				values.ConvertDurationMonths(2), true),
+			t: mustParseTime("1970-04-01T00:00:00Z"),
+			want: execute.Bounds{
+				Start: mustParseTime("1970-03-01T00:00:00Z"),
+				Stop:  mustParseTime("1970-08-01T00:00:00Z"),
+			},
+		},
```

Comment:
Could you add a few more test cases where the offset itself is negative? There seem to be specific code branches for the case where offset is negative and I want to make sure we are testing that code path.

Reply:
We already had one for nanoseconds, I added one for calendar durations.